### DISCUSSION TITLE
The Json Model of a dashboard from the UI is not the same as the one stored in the database or returned from an API call

### DISCUSSION THREAD

**Discussion by aengusrooneygrafana** [grafana/grafana#51385] [General]
The Json Model of a dashboard from the UI is not the same as the one stored in the database or returned from an API call

If you look at the end of two json files (one from the UI and one from the db/api), you can see that they both have the same uid and the same version number, but their schemaVersion is different.  The one from the UI is updated on the fly to the latest schemaVersion, and the one from the database has the schemaVersion from when it was last saved.

I have seen on this page (https://github.com/grafana/grafana/issues/10382 ) that there is a script (DashboardMigrator.ts) which updates the schema on the fly in the UI but I haven't been able to find a way to get the updated schemaVersion of a Dashboard from an API call.  

Is it possible to do this using an API call?  We are looking for a way to update the schemaVersion of the dashboards which have an older schemaVersion without having to manually save each dashboards.

**Comment by aengusrooneygrafana** [comment]
I have since learnt that there is currently no way to upgrade all dashboards. The only way right now is to open and save them. There is a benefit to this, as upgrading and saving all would make downgrades impossible (without a database restore).  However we do see that having a way to load and migrate & save all dashboards would be useful, so I'm opening a new feature request to add this as an option.

**Comment by jacksongoode** [comment]
@aengusrooneygrafana Are you able to view the json file from the browser locally without looking at the json model page for the dashboard?

**Reply by jacksongoode** [reply]
I ask because I've had a related discussion point https://github.com/grafana/grafana/discussions/47836

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
